compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -313,6 +315,14 @@ impl<'a> Parser<'a> {
             let pat = self.parse_pat_with_range_pat(false, None)?;
             self.sess.gated_spans.gate(sym::box_patterns, lo.to(self.prev_token.span));
             PatKind::Box(pat)
+        } else if self.eat_keyword(kw::Const) {
```

Comment:
Any reason not to use `check_inline_const` from above? This would treat `let const =` as an `IdentifierPattern` and continue to suggest `let r#const =` to the user.

Reply:
Right, changed to that. It was me distracted :).

Reply:
This comment is still relevant. The reason you're not preserving the more informative error is that you unconditionally start parsing an inline const pattern after seeing `const`. If this `eat_keyword` succeeds, you should check for the presence of `{` and fall back to the old code path (maybe `can_be_ident_pat`, maybe something else) if it is absent.

Reply:
Ohh right, I forgot about changing this. Will do.